src/systems/dynamic_detachable_joint/DynamicDetachableJoint.cc

```diff
@@ -0,0 +1,386 @@
+    }
+  }
+}
+
+//////////////////////////////////////////////////
+bool DynamicDetachableJoint::OnServiceRequest(const gz::msgs::AttachDetachRequest &_req,
+                                              gz::msgs::AttachDetachResponse &_res)
+{
+  GZ_PROFILE("DynamicDetachableJoint::OnServiceRequest");
+
+  // Check if the request is valid
+  if (_req.child_model_name().empty() || _req.child_link_name().empty() )
+  {
+    _res.set_success(false);
+    _res.set_message("Invalid request: child_model_name and child_link_name must be set.");
+    return true;
+  }
+
+  if (_req.command().empty())
+  {
+    _res.set_success(false);
+    _res.set_message("Invalid request: command must be 'attach' or 'detach'.");
+    return true;
+  }
+
+   // If attach is requested
+   if (_req.command() == "attach")
+   {
+     if (this->isAttached) 
+     {
+       _res.set_success(false);
+      _res.set_message("Already attached to child model " + this->attachedChildModelName +
+                       " at link " + this->attachedChildLinkName + ".");
+       gzdbg << "Already attached" << std::endl;
+       return true;
+     }
+
+    // set the child model and link names from the request
+    this->childModelName = _req.child_model_name();
+    this->childLinkName  = _req.child_link_name();
+     this->OnAttachRequest(msgs::Empty());
+     _res.set_success(true);
+     _res.set_message("Attached to child model " + this->childModelName +
+                      " at link " + this->childLinkName + ".");
+   }
+
+   // If detach is requested
+   else if (_req.command() == "detach")
+   {
+     if (!this->isAttached)
+     {
+         _res.set_success(false);
+         _res.set_message(std::string("Detach request received for ")
+              + this->attachedChildModelName + "/" + this->attachedChildLinkName);
+         gzdbg << "Already detached" << std::endl;
+         return true;
+     }
+
+    // Validate that the request matches what is actually attached.
+    const auto &reqModel = _req.child_model_name();
+    const auto &reqLink  = _req.child_link_name();
+    if (reqModel != this->attachedChildModelName ||
+        reqLink  != this->attachedChildLinkName)
+    {
+      _res.set_success(false);
+      _res.set_message(
+        "Detach rejected: requested " + reqModel + "/" + reqLink +
+        " but currently attached to " + this->attachedChildModelName + "/" +
+        this->attachedChildLinkName + "."
+      );
+      return true;
+    }
+
+     this->OnDetachRequest(msgs::Empty());
+     _res.set_success(true);
+     _res.set_message("Detached from child model " + this->attachedChildModelName +
+                      " at link " + this->attachedChildLinkName + ".");
+   }
+
+   else
+   {
+     _res.set_success(false);
+     _res.set_message("Invalid command. Use 'attach' or 'detach'.");
+     return true;
+   }
+   return true;
+ }
+
+//////////////////////////////////////////////////
+void DynamicDetachableJoint::OnAttachRequest(const msgs::Empty &)
```

Comment:
Why are we passing an empty message for no reason? We don't actually need this method.